### DISCUSSION TITLE
Constructor function name vote

### DISCUSSION THREAD

**Discussion by dmkozh** [stellar/stellar-protocol#1507] [Core Advancement Proposals]
Vote for the name of the Soroban [constructor ](https://github.com/stellar/stellar-protocol/blob/50dde0611440d6dc562a33462e6ba5f1504b2753/core/cap-0058.md) function name:

- `__init`
- `__constructor`
- `__ctor`
- `__create`
- `__on_create`

Every contract that needs to be initialized on creation will have to define this function.

Please respond in the thread simply with the name of the one you are voting for.

**Comment by Myestery** [comment]
__constructor looks okay

**Comment by KodeSage** [comment]
this would be `__constructor`

**Comment by tupui** [comment]
`__init` 😃 (Could we use a DAO for the vote?)

**Comment by villadaniel750** [comment]
__constructor it's OK

**Comment by aolieman** [comment]
`__init`, for initialization logic

**Comment by ShinySyntax** [comment]
__constructor it's OK

**Comment by Olawale-22** [comment]
__create

**Comment by gip** [comment]
`__init`

**Comment by robertDurst** [comment]
`__init`

**Comment by silence48** [comment]
I'm gonna vote for __constructor

**Comment by mdb1710** [comment]
_init

**Comment by faytey** [comment]
__constructor

**Comment by brianebert** [comment]
__constructor

**Comment by esteblock** [comment]
__constructor
Its also called like that in Solidity.

**Comment by tolgayayci** [comment]
__constructor

**Comment by leighmcculloch** [comment]
`__create`

**Comment by greenarmor** [comment]
__init  short, easy typing.

**Comment by overcat** [comment]
In some languages, a `constructor` is more involved in the creation and memory allocation of an object, while an `init` method focuses more on setting up the initial state of the object after it has been created.

**Comment by KevinEverywhere** [comment]
__constructor

**Comment by gidson5** [comment]
__contructor

**Comment by Ghostrayu** [comment]
__constructor feels right

**Comment by markuspluna** [comment]
__init is the industry standard, I don't see a reason to use a different name and confuse people

**Reply by dmkozh** [reply]
Would you mind which languages/frameworks you're referring to here? Solidity uses `constructor()` and I'm only aware of `__init__` constructor method in Python (which, to be frank, has inspired my original proposal) out of a dozen popular languages (`constructor` is not used explicitly a lot, but that's what typically the name of the language concept is). I'm genuinely curious what else uses that name.

**Comment by AghastyGD** [comment]
__init short and looks better

**Comment by manny-uncharted** [comment]
__constructor looks okay

**Comment by koolamusic** [comment]
`__init` does it for me

**Comment by tupui** [comment]
Maybe a bit late, but why not `__new`? Since we are doing Rust. That would seem to follow more the language itself no?

**Comment by glanzz** [comment]
__constructor

**Comment by skaunov** [comment]
__init

**Comment by schierlm** [comment]
`__init`

**Comment by PaigeDAO** [comment]
__constructor looks okay

**Comment by dmkozh** [comment]
Thanks everyone for voting! I think we've had enough time to gather enough opinions. `__constructor` leads with 14 votes vs 9 votes for `__init`. I'll proceed with updating the CAP with `__constructor` name. While I'm relatively indifferent on the specific option to use, `__constructor` seems slightly beneficial thanks to consistency with Solidity, which is probably the most likely language that a smart contract developer would be familiar with. OTOH I'm pretty sure neither option can be discovered by the developers intuitively without looking at the docs due to the need to use `__` prefix, so probably the name doesn't matter too much in the end.
